### PR TITLE
bus/rs232/printer: Add generic Radio Shack printer option

### PR DESCRIPTION
**Pull request by tlindner**
Radio Shack serial printer's use the receive line as a status line. If it is not set low, printing software will hang indefinitely.

### CODE REVIEW

**Comment by cuavas** [comment]
Not calling the base `device_start` is an anti-pattern.

**Comment by tlindner** [comment]
You are correct. Adjusted.

**Comment by smf-** [comment]
Would it be simpler to initialize m_initial_rx_state in the constructor
rather than overriding device_start()?

IIRC the base initializer is set before the derived class initializer.

**Comment by tlindner** [comment]
I think I've responded to all issues.

File: src/devices/bus/rs232/printer.cpp [hunk 1]
```diff
@@ -37,6 +55,7 @@ ioport_constructor serial_printer_device::device_input_ports() const
 
 void serial_printer_device::device_start()
 {
+	save_item(NAME(m_initial_rx_state));
```
Comment (cuavas) [inline src/devices/bus/rs232/printer.cpp:58]: There’s no need to save this – it’s set on construction and never changed.

**Comment by tlindner** [comment]
I removed the save state. But kept the empty device_start() method, since it was there to begin with.

**Pull request merged**